Create InputGroup component

HDinger opened the pull request:
This creates a new component `InputGroup` which consists of: 

* A text field
* An trailing action shown next to it. This can either be an icon button or a clipboardCopyButton

<img width="560" alt="Bildschirmfoto 2023-11-24 um 11 32 48" src="https://github.com/opf/primer_view_components/assets/7457313/8c8ccba3-48cc-4bc0-b803-2e91a6619c8f">

https://community.openproject.org/projects/openproject/work_packages/50741/activity

akabiru commented on app/components/primer/open_project/input_group.rb:
```diff
@@ -0,0 +1,74 @@
+
+module Primer
+  module OpenProject
+    # Add a general description of component here
+    # Add additional usage considerations or best practices that may aid the user to use the component correctly.
+    # @accessibility Add any accessibility considerations
```
```suggestion
    # `InputGroup` is composed of a text field input with a trailing action
```

akabiru commented on app/components/primer/open_project/input_group.rb:
```diff
@@ -0,0 +1,74 @@
+      #
+      # To render an iconButton, call the `with_trailing_action_icon` method, which accepts the arguments accepted by <%= link_to_component(Primer::Beta::Octicon) %>.
+      renders_one :trailing_action, types: {
+        clipboard_copy_button: lambda { |**system_arguments, &block|
```
Nice application of polymorphic slots ✨

akabiru reviewed (commented):
Top work @HDinger , tests look great- only one remark

- [ ] Should we add `input_width` support in this PR? I couldn't find a reference for this in this PR

HDinger commented:
> Top work @HDinger , tests look great- only one remark
> 
>     * [ ]  Should we add `input_width` support in this PR? I couldn't find a reference for this in this PR

Good point, @akabiru 👍 I added the support for the `input_width`, and added it to the playground.

akabiru reviewed (approved):
🥇 Thank you! 👏🏾

HDinger commented:
Merging as the lint issue issue is coming from upstream
